### PR TITLE
CB-21005 Mark volumes to deletable/non-deletable based on instanceId or FQDN

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java [hunk 1]
```diff
@@ -433,17 +438,19 @@ private Resource updateDeleteVolumesFlag(boolean deleteVolumes, Resource volumeS
...
-            Set<String> instanceIdsToRepair) {
-        if (!ManualClusterRepairMode.DRY_RUN.equals(repairMode) && !instanceIdsToRepair.isEmpty()) {
+            Set<InstanceMetaData> instances) {
+        if (!ManualClusterRepairMode.DRY_RUN.equals(repairMode) && !instances.isEmpty()) {
             LOGGER.info("Repair mode is not a dry run, {}", repairMode);
-            Predicate<Resource> updateVolumesPredicate = inInstances(instanceIdsToRepair);
-            updateVolumesDeleteFlag(stack, updateVolumesPredicate, deleteVolumes);
+            Set<String> instanceIds = instances.stream().map(InstanceMetadataView::getInstanceId).collect(toSet());
+            Set<String> instanceFQDNs = instances.stream().map(InstanceMetadataView::getDiscoveryFQDN).collect(toSet());
+            updateVolumesDeleteFlag(stack, instanceIds, instanceFQDNs, deleteVolumes);
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java:453]: the logic here is almost the same as the logic in `markVolumesToNonDeletable`, can be extracted both into a method

**Review by horadla23 — COMMENTED** [review]
- UT for `ClusterRepairService` would be good
- one short sentence to the commit message should be added to highlight the relevance of this change

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java (hunk 1 above)
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java:453]: you are right, I will extract it

**Comment by sodre90** [comment]
Added UTs to cover the deletion flag logic

**Review by horadla23 — APPROVED** [review]

**Pull request merged**